adapters/adapter.go

```diff
@@ -9,32 +9,42 @@ import (
 	"time"
 )
 
+// Adapters connect prebid-server to a demand partner. Their primary purpose is to produce bids
+// in response to Auction requests.
 type Adapter interface {
+	// A name which uniquely identifies this adapter.
+	// This must not overlap with any other adapters in prebid-server.
 	Name() string
 	FamilyName() string
 	GetUsersyncInfo() *pbs.UsersyncInfo
+	// Produce bids which should be considered, given the auction params.
```

Comment:
Not sure: this defines what an adapter is... do you mean be clear that it makes actual calls to the server?

If so, I'm not sure about that. A caller shouldn't care _how_ the bids are produced... and specifying it does limit the implementations. For example, it would prevent them from cacheing some bids under the hood, if they decided that fast response times were more important than the latest info from their servers.

Of course... it might also be a _good_ idea to limit implementations, if you don't want bidders to do crazy things like that.

In this project, do you think the flexibility is a good thing or a bad one?